### PR TITLE
Add a confirmation popup for study/case creation based on env property

### PR DESCRIPTION
**Pull request by dbraquart**
The goal is to define this property/behaviour for demo env only.

required: https://github.com/gridsuite/commons-ui/pull/548

to locally test tou can report this update https://github.com/gridsuite/deployment/pull/423/files#diff-ebb7c425aef637da9112c73d393992d7bc8e67dbfa770948218672e44ae336c1R4
into https://github.com/gridsuite/gridexplore-app/blob/main/public/env.json

### CODE REVIEW

**Review by ayolab — APPROVED** [review]

File: src/hooks/useConfidentialityWarning.ts
Comment (Tristan-WorkGH) [inline src/hooks/useConfidentialityWarning.ts:1]: maybe to move to commons-ui?

File: src/hooks/useConfidentialityWarning.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useEffect, useState } from 'react';
+import { fetchEnv } from '@gridsuite/commons-ui';
+
+export const useConfidentialityWarning = () => {
```
Comment (Tristan-WorkGH) [inline src/hooks/useConfidentialityWarning.ts]:
```suggestion
export const useConfidentialityWarningKey = () => {
```

File: src/components/dialogs/create-study-dialog/create-study-dialog.jsx [hunk 2]
```diff
@@ -60,6 +61,7 @@ const CreateStudyDialog = ({ open, onClose, providedExistingCase }) => {
     const intl = useIntl();
     const { snackError } = useSnackMessage();
     const dispatch = useDispatch();
+    const confidentialityWarning = useConfidentialityWarning();
```
Comment (Tristan-WorkGH) [inline src/components/dialogs/create-study-dialog/create-study-dialog.jsx]:
```suggestion
    const confidentialityWarningKey = useConfidentialityWarning();
```

File: src/components/dialogs/create-case-dialog/create-case-dialog.tsx [hunk 3]
```diff
@@ -48,6 +49,7 @@ interface CreateCaseDialogProps {
 const CreateCaseDialog: React.FunctionComponent<CreateCaseDialogProps> = ({ onClose, open }) => {
     const dispatch = useDispatch<AppDispatch>();
     const { snackError } = useSnackMessage();
+    const confidentialityWarning = useConfidentialityWarning();
```
Comment (Tristan-WorkGH) [inline src/components/dialogs/create-case-dialog/create-case-dialog.tsx]:
```suggestion
    const confidentialityWarningKey = useConfidentialityWarningKey();
```

File: src/components/dialogs/create-case-dialog/create-case-dialog.tsx [hunk 4]
```diff
@@ -114,6 +116,7 @@ const CreateCaseDialog: React.FunctionComponent<CreateCaseDialogProps> = ({ onCl
             onClose={onClose}
             onSave={handleCreateNewCase}
             disabledSave={!isFormValid}
+            confirmationMessage={confidentialityWarning}
```
Comment (Tristan-WorkGH) [inline src/components/dialogs/create-case-dialog/create-case-dialog.tsx]:
```suggestion
            confirmationMessageKey={confidentialityWarningKey}
```

File: src/hooks/useConfidentialityWarning.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    const [confidentialityWarning, setConfidentialityWarning] = useState<string>();
```
Comment (Tristan-WorkGH) [inline src/hooks/useConfidentialityWarning.ts]:
```suggestion
    const [confidentialityWarning, setConfidentialityWarning] = useState<string>('');
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: src/hooks/useConfidentialityWarning.ts (hunk 1 above)
Comment (dbraquart) [reply src/hooks/useConfidentialityWarning.ts]: For the hook itself, it sounds better without 'Key'

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**